test/test_sendrecv.jl

```diff
@@ -106,4 +106,18 @@ rreq = nothing
 sreq = nothing
 gc()
 
+root = 0
+g = x -> x^2 + 2x - 1
+if MPI.Comm_rank(comm) == root + 1
+    done, f, stat = MPI.irecv( root, root+32, comm )
+    if done
```

Comment:
This code calls `irecv` once. If there is no message at this time, then it will never be received. You should probably call `irecv` in a loop until `done` is `true`.